lib/closure_tree/hierarchy_maintenance.rb

```diff
@@ -35,10 +35,13 @@ def _ct_before_save
     end
 
     def _ct_after_save
-      if changes[_ct.parent_column_name] || @was_new_record
+      as_5_1 = ActiveSupport.version >= Gem::Version.new('5.1.0')
+      changes_method = as_5_1 ? :saved_changes : :changes
```

Comment:
>  I'm just frustrated by the rails team for making changes like this. 

For the record, this change was not "oh this is better this way". The change in behavior of these methods was required to fix *dozens* of bugs, and significantly simplify the implementation of associations (weeding out even more bugs). It's not something that was taken lightly.